screens/leaderboard/FriendsLeaderboardScreen.tsx

```diff
@@ -0,0 +1,91 @@
+import { Ionicons } from '@expo/vector-icons';
+import { useCallback, useEffect, useState } from 'react';
+import { SafeAreaView, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import { useLazyGetFriendLeaderboardQuery, useLazyGetGlobalLeaderboardQuery } from '../../components/user/userApi';
+import LeaderboardList from './LeaderboardList';
+
+const DEFAULT_PAGE = 0;
+const DEFAULT_PAGE_SIZE = 10;
+
+const FriendsLeaderboardScreen = () => {
+  const [currentPage, setCurrentPage] = useState(DEFAULT_PAGE);
+  const [totalPageNum, setTotalPageNum] = useState(0);
+  const [isRefreshing, setIsRefreshing] = useState(false);
+  const [trigger, { data: leaderboard, isFetching, isError }] = useLazyGetFriendLeaderboardQuery();
+
+  const goToPreviousPage = () => {
+    if (currentPage > 0) {
+      // setCurrentPage((prevPage) => prevPage - 1);
+      trigger({ size: DEFAULT_PAGE_SIZE, page: currentPage - 1 });
+    }
+  };
+
+  const goToNextPage = () => {
+    // setCurrentPage((prevPage) => prevPage + 1);
+    if (currentPage < totalPageNum - 1) {
+      trigger({ size: DEFAULT_PAGE_SIZE, page: currentPage + 1 });
+    }
+  };
+
+  const onRefresh = useCallback(async () => {
+    setIsRefreshing(true);
+    await trigger({ size: DEFAULT_PAGE_SIZE, page: currentPage });
+    setIsRefreshing(false);
+  }, []);
+
+  useEffect(() => {
+    trigger({ size: DEFAULT_PAGE_SIZE });
+    // setCurrentPage(leaderboard!.currentPage);
+  }, []);
+
+  useEffect(() => {
+    if (leaderboard) {
+      setCurrentPage(leaderboard.currentPage);
+      setTotalPageNum(leaderboard.totalPages);
```

Comment:
Handling these stuff in a `useEffect` is not suggested. Also, we don't need to keep these as a separate state anyways. `leaderboard` variable already contains these information we can have these as variables.
Just add the following lines after `useLazyGetFriendLeaderboardQuery`:
```js
const currentPage = leaderboard?.currentPage || DEFAULT_PAGE;
const totalPageNum = leaderboard?.totalPages || 0;
```


And then remove the states for `currentPage` and `totalPageNum`. Also please remove the commented state changes.